server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/text/KeyboardKeyEvent.java

```diff
@@ -0,0 +1,53 @@
+package sh.calaba.instrumentationbackend.actions.text;
+
+import android.text.Selection;
+import android.view.KeyEvent;
+import android.view.inputmethod.InputConnection;
+
+import sh.calaba.instrumentationbackend.InstrumentationBackend;
+import sh.calaba.instrumentationbackend.Result;
+import sh.calaba.instrumentationbackend.actions.Action;
+import sh.calaba.instrumentationbackend.actions.softkey.KeyUtil;
+
+public class KeyboardKeyEvent implements Action {
```

Comment:
This is not the same as adb shell input keyevent I think. This is sending keyboard inputs to the input connection, which may react differently